pkg/gke/components/CruGKE.vue

```diff
@@ -0,0 +1,840 @@
+          if (!this.isAuthenticated) {
+            return;
+          }
+          const valid = (p: GKENodePool) => (p.autoscaling.minNodeCount || 0) <= (p.autoscaling.maxNodeCount || 0);
+          let msg = null;
+
+          this.nodePools.forEach((p) => {
+            if (!valid(p)) {
+              p._minMaxValid = false;
+              msg = this.t('gke.errors.minMaxNodeCount');
+            } else {
+              p._minMaxValid = true;
+            }
+          });
+
+          return msg;
+        },
+
+        nodePoolNameUnique: () => {
+          if (!this.isAuthenticated) {
+            return;
+          }
+          let msg = null;
+          const allPoolNames: string[] = this.nodePools.map((p) => p.name);
+
+          this.nodePools.forEach((p) => {
+            if (allPoolNames.filter((n) => n === p.name).length > 1) {
+              p._nameUnique = false;
+              msg = this.t('gke.errors.poolNamesUnique');
+            } else {
+              p._nameUnique = true;
+            }
+          });
+
+          return msg;
+        },
+
+        poolNameRequired: (name:string) => {
+          if (!this.isAuthenticated) {
+            return;
+          }
+          let msg = null;
+          const valid = (n: string) => !!n;
+
+          if (name || name === '') {
+            return !valid(name) ? this.t('validation.required', { key: this.t('gke.groupName.label') }) : null;
+          }
+          this.nodePools.forEach((p) => {
+            if (!valid(p.name)) {
+              msg = this.t('validation.required', { key: this.t('gke.groupName.label') });
+            }
+          });
+
+          return msg;
+        }
+
+      };
+    },
+
+    // upstreamSpec will be null if the user created a cluster with some invalid options such that it ultimately fails to create anything in gke
+    // this allows them to go back and correct their mistakes without re-making the whole cluster
+    isNewOrUnprovisioned() {
+      return this.mode === _CREATE || !this.normanCluster?.gkeStatus?.upstreamSpec;
+    },
+
+    isEdit() {
+      return this.mode === _CREATE || this.mode === _EDIT;
+    },
+
+    doneRoute() {
+      return this.value?.listLocation?.name;
+    },
+
+    hasCredential() {
+      return !!this.config?.googleCredentialSecret;
+    },
+
+    clusterId(): String | null {
+      return this.value?.id || null;
+    },
+
+    canManageMembers(): Boolean {
+      return canViewClusterMembershipEditor(this.$store);
+    },
+
+    CREATE(): string {
+      return _CREATE;
+    },
+
+    VIEW(): string {
+      return _VIEW;
+    },
+
+    machineTypeOptions(): {label: string, kind?: string, value?: string, disabled?: boolean, [key: string]: any}[] {
+      const allTypes = this.machineTypesResponse?.items;
+      const out = [] as {label: string, kind?: string, value?: string, disabled?: boolean, [key: string]: any}[];
+
+      if (!allTypes) {
+        return out;
+      }
+
+      allTypes.forEach((type, i) => {
```

Comment:
TS needs some typin' here, line 532 (`this.errors.push(err);`) and lines 548, 549 (`this.$refs.pools?.select` related)

Reply:
Intellisense thinks this _is_ typed, because `machineTypesResponse` is typed. Is there a benefit to explicitly calling out the type here as well?

![Screenshot 2024-06-10 at 1 51 51 PM](https://github.com/rancher/dashboard/assets/42977925/57b18251-7621-43d6-9071-dff5ec07e543)

I updated typing on `this.errors` and `this.$refs.pools?.select`...`errors` still throws errors in the template - it's an awkward one to type because its coming from a mixin, which simply do not play well with typescript. I looked into it previously and didn't find a nice clean solution for the template errors.

Reply:
I think we might have hit the same issue in a previous PR, looked into it a bit more. With this one i don't get the typing for `type` because of the imported type files end with `.d.ts`. Removing those from `types/gcp` and `types/index` solved the issue, however exposed an issues
- line 491 `name:     group,` (name is missing from `GKEMachineTypeOption`)

which just leaves two, both mixins and can be ignored
- pools.select 
- emitOrRoute

Reply:
there's a couple of other references to .d.ts in the plugin as well

Reply:
Given the update to my now standard vs code extension i don't get these